glommio/src/executor/mod.rs

```diff
@@ -2132,9 +2138,38 @@ impl ExecutorProxy {
     /// and should be preferred over unconditional yielding methods like
     /// [`ExecutorProxy::yield_now`] and
     /// [`ExecutorProxy::yield_task_queue_now`].
-    #[inline]
+    #[inline(always)]
     pub async fn yield_if_needed(&self) {
-        Self::cond_yield(|local_ex| local_ex.need_preempt()).await;
+        #[cfg(not(feature = "native-tls"))]
+        {
+            let need_yield = if LOCAL_EX.is_set() {
+                LOCAL_EX.with(|local_ex| {
+                    if local_ex.need_preempt() {
+                        local_ex.mark_me_for_yield();
+                        true
+                    } else {
+                        false
+                    }
+                })
+            } else {
+                // We are not in a glommio context
+                false
+            };
+
+            if need_yield {
+                futures_lite::future::yield_now().await;
+            }
+        }
+
+        #[cfg(feature = "native-tls")]
+        unsafe {
+            if !LOCAL_EX.is_null() {
```

Comment:
and if you tackle the above to just return false when the executor is null, then you don't need this test here

Reply:
I did that too. I am not seeing any measurable improvement but it makes the code shorter so I'll take it.